core/c4diagram.go

```diff
@@ -33,7 +33,13 @@ func (c *clientPlantUML) Do(v DiagramGraph) ([]byte, error) {
 	if err != nil {
 		return nil, err
 	}
-	return c.requestHandler("svg/" + p)
+
+	diagram, err := c.requestHandler("svg/" + p)
+	if err != nil {
+		return nil, err
+	}
+
+	return ResponseC4Diagram{SVG: *(*string)(unsafe.Pointer(&diagram))}, nil
```

Comment:
Please note that this "trick" is meant to spare a single memory allocation, which in the context may mean a lot given the size of an svg output object. It may be considered premature optimisation, but it has low implementation costs - I went for it :) WDYT?

For the context, the alternative is:
```go
return ResponseC4Diagram{SVG: string(diagram)}, nil
```

It's pro: readability and maintenance cost; con: additional memory allocation, runtime etc.

Reply:
🤔 Not sure... `unsafe` is not protected by the Go compatibility guide and it might introduce some subtle bugs that are very hard to debug, but as you said, we might face some challenges regarding resources consumptions due to the size of these SVG outputs. So, if the `go vet`didn't identificate a possible misuse of `unsafe`and since it is one of the few exceptions that we can use `unsafe` to write efficient code, I'm fine.

Reply:
You are very much to the point with your reasoning! Thank you!

Reply:
For clarity

> unsafe is not protected by the Go compatibility guide

`unsafe` is used by many standard packages, so I would not consider it a significant risk factor ;)